engine/throttle.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Limits the concurrency of the `fn`.
+ * @param {Number} concurrency - maxium number of inflight calls to `fn`
+ * @param {Function} fn - function that returns a promise.
+ * @return {Promise}
+ */
+export default function throttle(concurrency, fn) {
```

Comment:
There are, two I tried were broken with our node version or didn't use built in promises and were broken. Just found another that seems to be more up to date https://www.npmjs.com/package/bottleneck. I'll give that whirl.